Neos.ContentRepository.Core/Classes/SharedModel/Node/NodeAddress.php

```diff
@@ -0,0 +1,106 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Neos\ContentRepository\Core\SharedModel\Node;
+
+use Neos\ContentRepository\Core\DimensionSpace\DimensionSpacePoint;
+use Neos\ContentRepository\Core\Projection\ContentGraph\Node;
+use Neos\ContentRepository\Core\SharedModel\ContentRepository\ContentRepositoryId;
+use Neos\ContentRepository\Core\SharedModel\Workspace\WorkspaceName;
+
+/**
+ * This describes a node's read model identity namely:
+ *
+ * - {@see ContentRepositoryId}
+ * - {@see WorkspaceName}
+ * - {@see DimensionSpacePoint} (not to be confused with the {@see Node::$originDimensionSpacePoint})
+ * - {@see NodeAggregateId}
+ *
+ * In combination the parts can be used to distinctly identify a single node.
+ *
+ * By using the content graph for the content repository
+ * one can build a subgraph with the right perspective to find this node:
+ *
+ *      $subgraph = $contentRepository->getContentGraph($nodeAddress->workspaceName)->getSubgraph(
+ *          $nodeAddress->dimensionSpacePoint,
+ *          VisibilityConstraints::withoutRestrictions()
+ *      );
+ *      $node = $subgraph->findNodeById($nodeAddress->aggregateId);
+ *
+ * @api
+ */
+final readonly class NodeAddress implements \JsonSerializable
```

Comment:
In terms of ease of use I would totally like a NodeAddress::resolveNode(ContentRepositoryRegistry) for example ;)

Reply:
passing the `ContentRepositoryRegistry` is illegal and we have - yet - no `ContentRepositoryRegistryInterface` which would allow this.
I would delay this discussion for now or move it over to https://github.com/neos/neos-development-collection/pull/5047

I generally agree with you that we need nice utility methods i think but currently we quite not know yet what they should look like.